moveit_core/planning_request_adapter/src/planning_request_adapter.cpp

```diff
@@ -140,24 +142,28 @@ bool PlanningRequestAdapterChain::adaptAndPlan(const planning_interface::Planner
       fn = [&adapter = *adapters_[i], fn, &added_path_index = added_path_index_each[i]](
                const planning_scene::PlanningSceneConstPtr& scene, const planning_interface::MotionPlanRequest& req,
                planning_interface::MotionPlanResponse& res) {
-        return callAdapter(adapter, fn, scene, req, res, added_path_index);
+        // Abort pipeline and return in case of failure
+        callAdapter(adapter, fn, scene, req, res, added_path_index);
+        return (res.error_code_.val == moveit_msgs::msg::MoveItErrorCodes::SUCCESS);
```

Comment:
Here you are returning a bool, even though the return type should be MoveItErrorCode. Why can't all functions (`callAdapter()` etc) return a MoveItErrorCode, and once you detect a non-successful error anywhere in the pipeline, you return that as result. No need to pass in a mutable if you fail early and pass back the result in the pipeline function chain.